addons/web/controllers/main.py

```diff
@@ -2084,10 +2084,13 @@ def report_barcode(self, type, value, width=600, height=100, humanreadable=0, qu
         :param mask: The mask code to be used when rendering this QR-code.
                      Masks allow adding elements on top of the generated image,
                      such as the Swiss cross in the center of QR-bill codes.
+        :param kwargs: Additional parameters for the barcode creation.
+                       Now used: 'barLevel': QR code Error Correction Levels. Default is 'L'.
+                       ref: https://github.com/mattjmorrison/ReportLab/tree/master/src/reportlab/graphics/barcode/qr.py#L62
         """
         try:
             barcode = request.env['ir.actions.report'].barcode(type, value, width=width,
-                height=height, humanreadable=humanreadable, quiet=quiet, mask=mask)
+                height=height, humanreadable=humanreadable, quiet=quiet, mask=mask, kwargs=kwargs)
```

Comment:
GIven the defaults match, every strictly keyword parameter could just be passed as a kwarg in bulk.